tools/contrib/fsurdat_modifier.py

```diff
@@ -0,0 +1,356 @@
+ 
+To remove NPL (ncar_pylib) from your environment on Cheyenne/Casper:
+    deactivate
+-------------------------------------------------------------------
+
+"""
+
+#  Import libraries
+from __future__ import print_function
+
+import sys
+import os
+import string
+import logging
+import subprocess
+import argparse
+
+import xarray as xr
+
+from datetime import date
+from getpass import getuser
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+
+myname = getuser()
+
+def get_parser():                                                                                                                                                                   
+        """
+        Get parser object for this script.
+
+        Command-line inputs
+        - fsurdat_in: input file (str)
+        - fsurdat_out: output file (str)
+        - variable: variable to modify (str)
+        - value: value assigned to the variable to be modified (float)
+        """
+        parser = ArgumentParser(description=__doc__,
+                               formatter_class=argparse.RawDescriptionHelpFormatter)
+
+        parser.print_usage = parser.print_help
+
+        parser.add_argument('--dom_pft', 
+                    help='Dominant PFT if overwrite_single_pft = .true. [default: %(default)s] ', 
+                    action="store", 
+                    dest="dom_pft",
+                    type =int,
+                    default=7)
+        parser.add_argument('--uni_snow', 
+                    help='Turn on the flag to create uniform snowpack. [default: %(default)s]', 
+                    action="store", 
+                    dest="uni_snow",
+                    default=False)
+        parser.add_argument('--overwrite_single_pft', 
+                    help='Turn on the flag to make the whole grid 100%% single PFT. [default: %(default)s]', 
+                    action="store", 
+                    dest="overwrite_single_pft",
+                    default=False)
+        parser.add_argument('--zero_nonveg', 
+                    help='Set all non-vegetation landunits to zero. [default: %(default)s]', 
+                    action="store", 
+                    dest="zero_nonveg",
+                    type =bool,
+                    default=False)
+        parser.add_argument('--no_saturation_excess', 
+                    help='Turn off saturation excess. [default: %(default)s]', 
+                    action="store", 
+                    dest="no_saturation_excess",
+                    type =bool,
+                    default=False)
+        parser.add_argument('--fsurdat_in', 
+                    help = 'Input surface dataset. [default: %(default)s]', 
+                    action = "store", 
+                    dest = "fsurdat_in",
+                    type = str,
+                    default = "/glade/p/cesmdata/cseg/inputdata/lnd/clm2/surfdata_map/release-clm5.0.18/surfdata_0.9x1.25_hist_78pfts_CMIP6_simyr2000_c190214.nc")
```

Comment:
- [ ] Again, for this and fsurdat_out, get rid of the default so it needs to be explicitly set